### PR TITLE
edit: make native-tls optional

### LINKED ISSUE #12: Replace native-tls with rustls
### Check Before Asking

- [X] Please check the issue list and confirm this feature is encountered for the first time.
- [X] Please try full text in English and attach precise description.

### Description

Follow-up request of https://github.com/CeresDB/ceresdb/issues/662.

Simply put, I don't want CeresDB to depend on openssl, so I suggest replace it with https://github.com/rustls/rustls

There is only one dependency in this crates which depends on openssl: reqwest, it has rustls support already, we just need to enable it.

### Other Information

Similar changes: https://github.com/NoXF/oss-rust-sdk/pull/17

### PR DESCRIPTION
**Pull request by dust1**
### What changes were proposed in this pull request?

Closes #12 

edit Cargo.toml to make native-tls optional

### Why are the changes needed?

native-tls is a heavy dep, which depend on openssl on local system, which may cause issue when build host and target host have different openssl version.

### Will break the compatibility? How if so?

none

### Does this PR introduce any user-facing change?
none

### How was this patch tested?

```
➜  obkv-table-client-rs git:(issue12) cargo tree -e features -i reqwest --no-default-features --features rustls-tls
reqwest v0.11.14
├── reqwest feature "__rustls"
│   └── reqwest feature "rustls-tls-webpki-roots"
│       └── reqwest feature "rustls-tls"
│           └── obkv-table-client-rs feature "rustls-tls" (command-line)
├── reqwest feature "__tls"
│   └── reqwest feature "__rustls" (*)
├── reqwest feature "blocking"
│   └── obkv-table-client-rs v0.1.0 (/Users/kous/myProjects/rustproject/obkv-table-client-rs)
│       ├── obkv-table-client-rs feature "reqwest"
│       │   └── obkv-table-client-rs feature "rustls-tls" (command-line)
│       └── obkv-table-client-rs feature "rustls-tls" (command-line)
├── reqwest feature "hyper-rustls"
│   └── reqwest feature "__rustls" (*)
├── reqwest feature "rustls"
│   └── reqwest feature "__rustls" (*)
├── reqwest feature "rustls-pemfile"
│   └── reqwest feature "__rustls" (*)
├── reqwest feature "rustls-tls" (*)
├── reqwest feature "rustls-tls-webpki-roots" (*)
├── reqwest feature "tokio-rustls"
│   └── reqwest feature "__rustls" (*)
└── reqwest feature "webpki-roots"
    └── reqwest feature "rustls-tls-webpki-roots" (*)
```

### Checklist

- [x] I've run the tests to see all new and existing tests pass.
- [x] If this Pull Request resolves an issue, I linked to the issue in the text above.
- [x] I've informed the technical writer about the documentation change if necessary.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -28,7 +28,7 @@ prometheus = "0.7"
 quick-error = "1.2"
 r2d2 = "0.8.3"
 rand = "0.6"
-reqwest = "0.9"
+reqwest = { version = "0.11.13", features = ["blocking"], default-features = false, optional = true }
```
[the remark this answers is not in the document]
Comment (dust1) [reply Cargo.toml:31]: Sorry I didn't check it carefully, I will resubmit it later

**Comment by dust1** [comment]
It can be compiled.
```
┌──(yoke㉿DESKTOP-2)-[~/temp/obkv-table-client-rs]
└─$ cargo build
    Finished dev [unoptimized + debuginfo] target(s) in 0.32s
warning: the following packages contain code that will be rejected by a future version of Rust: rustc-serialize v0.3.24
note: to see what the problems were, use the option `--future-incompat-report`, or run `cargo report future-incompatibilities --id 1`
```

File: Cargo.toml [hunk 2]
```diff
@@ -37,7 +37,7 @@ serde_derive = "1.0"
 serde_json = "1.0"
 spin = "0.5.0"
 tokio-codec = "0.1"
-uuid = "0.7"
+uuid = {version = "1.3.0", default-features = false, features = ["v4", "fast-rng", "macro-diagnostics"]}
```
Comment (ShiKaiWi) [inline Cargo.toml:40]:
```suggestion
uuid = { version = "1.3.0", default-features = false, features = ["v4", "fast-rng", "macro-diagnostics"] }
```

**Comment by ShiKaiWi** [comment]
> It can be compiled.
> 
> ```
> ┌──(yoke㉿DESKTOP-2)-[~/temp/obkv-table-client-rs]
> └─$ cargo build
>     Finished dev [unoptimized + debuginfo] target(s) in 0.32s
> warning: the following packages contain code that will be rejected by a future version of Rust: rustc-serialize v0.3.24
> note: to see what the problems were, use the option `--future-incompat-report`, or run `cargo report future-incompatibilities --id 1`
> ```

Just a tiny style issue. After fixing, @IHEII could you approve this PR?

**Review by ShiKaiWi — APPROVED** [review]

**Review by chunshao90 — APPROVED** [review]
LGTM

**Pull request merged**